### ISSUE TITLE
Twitch compatibility

### ISSUE THREAD

**Issue by Damianuscz** [MCMrARM/revolution-irc#169]
Hello,

First thx for super irc client. But i have a question about compatibility with twitch irc server. When i connect on server in to the room i can not see user.

**Comment by MCMrARM** [comment]
Screenshot?

**Comment by Damianuscz** [comment]
https://ibb.co/bVHnoL
https://ibb.co/mWNbhf
https://ibb.co/cPOcoL

**Comment by MCMrARM** [comment]
Well I'm unsure what the hell is happening in there but the channel is not prefixed with a # like it should be.

**Comment by Damianuscz** [comment]
yep. but if i use # to connect room server only write message from second screen "end of ...... "

**Comment by MCMrARM** [comment]
Just wait a day or two, 0.5.1 release is coming :P

**Comment by Damianuscz** [comment]
Still same problem. Users list i can not see on twitch server.

**Comment by MCMrARM** [comment]
Well, I think the Twitch server does not send the member list. Can you please test it on a different client?

**Comment by Damianuscz** [comment]
I do not have problem with hexchat

**Comment by MCMrARM** [comment]
Can you please try to manually run `/LIST #channelname`?

**Comment by MCMrARM** [comment]
I'll look into it later.

**Comment by Damianuscz** [comment]
List command = unknown command

**Comment by Phoenix616** [comment]
That API is not standard IRC behaviour. Would be interesting to see how HexChat supports it as I haven't seen an IRC client being able to do that purely with Twitch's IRC compatibility layer yet.

**Comment by Damianuscz** [comment]
Yes you have right. But user list is only one obstacle to use revolution irc for greatest mainstream irc network of today. If MCMrARM implement user list from twitch api is allready done.

**Comment by michcioperz** [comment]
Knowing him I don't think he would be cool with bending to the will of Twitch because that's just not IRC if it doesn't implement basic IRC commands.

**Comment by Damianuscz** [comment]
You have right! I dont know how twitch work inside. Maybe over bridge to irc maybe customized irc server. But we have oportuninty to implement small solution for big number of people. Otherside twitch irc works only like chat so no needs other thinks like dcc etc.. Last think to do is user list. Future whisper implementation or twitch emoty. But now for full function just user list.

**Comment by Phoenix616** [comment]
Maybe a possible workaround for this without adding service-specific functions could be to at least add all known users (the ones who chatted) to the user list if they aren't already in it. That way you can at least see who you talked with and maybe complete their names?

**Comment by MCMrARM** [comment]
Why don't you go and complain to Twitch instead seriously. Why should this project implement workarounds for clearly broken IRCd-s?

**Comment by Phoenix616** [comment]
Because people have and Twitch (which also is Amazon now) does usually not
listen to user complaints, especially if they are for deprecated/abandoned
features that are only kept working as a courtesy.

Adding a minor workaround for services that have these issues in a client
looks therefore more fruitful. (But I'm not requesting depending on their
API, I'm only interested in adding better support for networks that don't
send the full user list which tbh. could also be done to hide users in IRC
channels)

**Comment by MCMrARM** [comment]
Well in any case, I'm not going to add a workaround until I officially add scripting support. A pull request might be accepted if it's sane enough, but it won't be accepted if it calls into Twitch API.
